libs/providers/flagd/src/lib/service/http/service.ts

```diff
@@ -103,7 +103,7 @@ export class HTTPService implements Service {
   async resolveObject<T extends object>(flagKey: string, defaultValue: T, context: EvaluationContext): Promise<ResolutionDetails<T>> {
     try {
       const res = await axios.post<ResolveObjectResponse>(
-        `${this.url}/flags/${flagKey}/resolve/object`,
+        encodeURI(`${this.url}/flags/${flagKey}/resolve/object`),
```

Comment:
I think you need to do this everywhere you are constructing URLs for safety. If a flag name had a `/` in it, it would break. TBH I would just use `encodeURIComponent` on just the flag name.

Reply:
I have made this change, however after testing with flagd it seems that keys specifically containing ```/``` will still be parsed wrong by grpc-router, I will investigate the cause of this and open an issue.

Reply:
It might be better to update the schema to restrict the characters that are allowed to be used as the key.

https://github.com/open-feature/schemas/blob/main/json/flagd-definitions.yaml

We could restrict it to something like `a-z`, `A-Z`, `0-9`, `_`, and `-`.

Reply:
> I have made this change, however after testing with flagd it seems that keys specifically containing / will still be parsed wrong by grpc-router, I will investigate the cause of this and open an issue.

> It might be better to update the schema to restrict the characters that are allowed to be used as the key.

Yep, I'm OK with this too. I did verify the HTTP framework decoded path params containing encoded chars properly, but it also had trouble with `/` specifically (even encoded). I'm fine with not supporting that char.